components/server/src/memory.rs

```diff
@@ -13,15 +13,32 @@ pub struct MemoryTraceManager {
 
 impl MemoryTraceManager {
     pub fn flush(&mut self, _now: Instant) {
+        let raft_router = "raft_router".to_string();
+        let apply_router = "apply_router".to_string();
         for provider in &self.providers {
             let provider_name = provider.name();
             let ids = provider.get_children_ids();
             for id in ids {
                 let sub_trace = provider.sub_trace(id);
                 let sub_trace_name = sub_trace.name();
-                MEM_TRACE_SUM_GAUGE
-                    .with_label_values(&[&format!("{}-{}", provider_name, sub_trace_name)])
-                    .set(sub_trace.sum() as i64)
+                if sub_trace_name == raft_router || sub_trace_name == apply_router {
+                    let alive = sub_trace.sub_trace(Id::Name("alive"));
+                    let leak = sub_trace.sub_trace(Id::Name("leak"));
+                    MEM_TRACE_SUM_GAUGE
+                        .with_label_values(&[&format!(
+                            "{}-{}-alive",
+                            provider_name, sub_trace_name
+                        )])
+                        .set(alive.sum() as i64);
+
+                    MEM_TRACE_SUM_GAUGE
+                        .with_label_values(&[&format!("{}-{}-leak", provider_name, sub_trace_name)])
```

Comment:
`with_label_values` may acquire a lock here, is `flush` called frequently?

Reply:
FYI, it's 1 second-level timer for flushing this metrics.
https://github.com/tikv/tikv/blob/3ae1fb4320737c71a1c9d3f8ee6a3b7a9af6f6ea/components/server/src/server.rs#L225
Maybe should be longer for this update?

Reply:
I think it's ok. No reason to update it.